traits/trait_types.py

```diff
@@ -1700,127 +1701,263 @@ def __init__(
-                high = float(high)
-
-        elif vtype is int:
-            self._validate = "int_validate"
-            kind = 3
-            self._type_desc = "an integer"
-            if low is not None:
-                low = int(low)
-
-            if high is not None:
-                high = int(high)
-        else:
-            self.get, self.set, self.validate = self._get, self._set, None
-            self._vtype = None
-            self._type_desc = "a number"
-
-            if isinstance(high, six.string_types):
-                self._high_name = high = "object." + high
-            else:
-                self._vtype = type(high)
-            high = compile(str(high), "<string>", "eval")
+        if high_name is not None and high is not None:
+            raise TraitError(
+                "At most one of *high* and *high_name* should be provided."
+            )
 
-            if isinstance(low, six.string_types):
-                self._low_name = low = "object." + low
-            else:
-                self._vtype = type(low)
-            low = compile(str(low), "<string>", "eval")
+        value_trait = trait_from(value_trait)
 
-            if isinstance(value, six.string_types):
-                value = "object." + value
-            self._value = compile(str(value), "<string>", "eval")
+        super(BaseRange, self).__init__(**metadata)
 
-            self.default_value_type = CALLABLE_DEFAULT_VALUE
-            self.default_value = self._get_default_value
+        self.default_value_type = CALLABLE_DEFAULT_VALUE
+        self.default_value = self._get_default_value
 
-        exclude_mask = 0
-        if exclude_low:
-            exclude_mask |= 1
+        # For clip_on_get, need to override get and set.
+        if clip_on_get:
+            self.get = self._get
+            self.set = self._set
+            self.validate = None
 
-        if exclude_high:
-            exclude_mask |= 2
+        can_use_fast_validation = (
+            not clip_on_get
+            and low_name is high_name is None
+            and isinstance(value_trait.trait_type, Float)
+            and (low is None or isinstance(low, float))
+            and (high is None or isinstance(high, float))
+        )
 
-        if is_static and (vtype is not int):
-            self.init_fast_validator(kind, low, high, exclude_mask)
+        if can_use_fast_validation:
+            self.init_fast_validator(
+                4,
+                None if low is None else float(low),
+                None if high is None else float(high),
+                bool(exclude_low) | bool(exclude_high) << 1,
+            )
 
         #: Assign type-corrected arguments to handler attributes:
         self._low = low
         self._high = high
+        self._value = value
         self._exclude_low = exclude_low
         self._exclude_high = exclude_high
+        self._value_trait = value_trait
+        self._low_name = low_name
+        self._high_name = high_name
```

Comment:
Not sure how we feel about public vs. private APIs between Traits and TraitsUI, but I think that the `RangeEditor` is going to want access to these, so maybe these attributes (and possibly others) should be public?